https/tls_config.go

```diff
@@ -108,15 +128,41 @@ func ConfigToTLSConfig(c *TLSStruct) (*tls.Config, error) {
 }
 
 // Listen starts the server on the given address. If tlsConfigPath isn't empty the server connection will be started using TLS.
-func Listen(server *http.Server, tlsConfigPath string) error {
+func Listen(server *http.Server, tlsConfigPath string, logger log.Logger) error {
 	if (tlsConfigPath) == "" {
 		return server.ListenAndServe()
 	}
-	var err error
-	server.TLSConfig, err = getTLSConfig(tlsConfigPath)
-	if err != nil {
+
+	if err := validateUsers(tlsConfigPath); err != nil {
 		return err
 	}
+
+	// Setup basic authentication.
+	var handler http.Handler = http.DefaultServeMux
+	if server.Handler != nil {
+		handler = server.Handler
+	}
+	server.Handler = &userAuthRoundtrip{
+		tlsConfigPath: tlsConfigPath,
+		logger:        logger,
+		handler:       handler,
+	}
+
+	config, err := getTLSConfig(tlsConfigPath)
+	switch err {
+	case nil:
+		// Valid TLS config.
+	case errNoTLSConfig:
+		// No TLS config, back to plain HTTP.
+		level.Info(logger).Log("msg", "A web config file has been provided without TLS settings. TLS can't be enabled on the fly.")
```

Comment:
cannot

Should we also have an error in the other direction?

Reply:
In the other direction we would send a 500 error if the TLS config is gone from the file.

Reply:
I was wondering more in terms of log messages.

Reply:
Now I log a message in every case.